src/models/userStorage.ts

```diff
@@ -12,17 +13,48 @@ export class UserStorage {
     let user = UserStorage.users.find((u) => u.userId === profile.id);
     const { id, displayName, mail, givenName, surname } = profile._json;
 
-    // console.log("Microsoft Profile Data:", profile);
-
     if (user) {
       done(null, user);
     } else {
+      const picture = await getProfilePicture(accessToken);
       user = new User(id, displayName, mail)
         .withName(givenName, surname)
         .withAdminStatus(false)
-        .withProvider("Microsoft");
+        .withProvider("Microsoft")
+        .withPicture(picture);
       UserStorage.users.push(user);
       done(null, user);
     }
   }
 }
+
+export async function getProfilePicture(accessToken: string): Promise<string> {
+  console.log(`The access token: ${accessToken}`);
+  try {
+    const url = "https://graph.microsoft.com/v1.0/me/photo/$value";
+    const response = await fetch(url, {
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+      },
+    });
+
+    if (!response.ok) {
+      throw new Error(`Failed to fetch profile picture: ${response.status}`);
+    }
+
+    const arrayBuffer = await response.arrayBuffer();
+
+    if (arrayBuffer.byteLength === 0) {
+      return "";
+    }
+
+    const buffer = Buffer.from(arrayBuffer);
+
+    const dataURL = `data:image/jpeg;base64,${buffer.toString("base64")}`;
```

Comment:
It might be a nice idea to save the content to a cached file (e.g. name == <user.id>.jpeg) and serve it via another endpoint. For now a dataURL is of course fine.
A specific file might become handy, if you have a user list and want to display the picture of each user.

Reply:
Good thinking!

Reply:
https://github.com/inspired-consulting/profiles/issues/28